yarpl/test/Observable_test.cpp

```diff
@@ -285,6 +289,73 @@ TEST(Observable, toFlowableWithCancel) {
   EXPECT_EQ(v, std::vector<int>({1, 2, 3, 4, 5}));
 }
 
+TEST(Observable, toFlowableErrorStrategy) {
+  auto a = Observables::range(1,10);
+  auto f = a->toFlowable(BackpressureStrategy::ERROR);
+
+  std::vector<int64_t> v;
+
+  auto subscriber = make_ref<testing::StrictMock<yarpl::flowable::MockSubscriber<int64_t>>>(5);
+
+  EXPECT_CALL(*subscriber, onSubscribe_(_));
+  EXPECT_CALL(*subscriber, onNext_(_))
+      .WillRepeatedly(Invoke([&](int64_t value) {
+          v.push_back(value);
+      }));
+  EXPECT_CALL(*subscriber, onError_(_))
+      .WillOnce(Invoke([&](folly::exception_wrapper ex) {
+          EXPECT_TRUE(ex.is_compatible_with<yarpl::flowable::MissingBackpressureException>());
+      }));
+
+  f->subscribe(subscriber);
+
+  EXPECT_EQ(v, std::vector<int64_t>({1, 2, 3, 4, 5}));
+}
+
+TEST(Observable, toFlowableBufferStrategy) {
+  auto a = Observables::range(1,10);
+  auto f = a->toFlowable(BackpressureStrategy::BUFFER);
+
+  std::vector<int64_t> v;
+
+  auto subscriber = make_ref<testing::StrictMock<yarpl::flowable::MockSubscriber<int64_t>>>(5);
+
+  EXPECT_CALL(*subscriber, onSubscribe_(_));
+  EXPECT_CALL(*subscriber, onNext_(_))
+      .WillRepeatedly(Invoke([&](int64_t value) {
+          v.push_back(value);
+      }));
+  EXPECT_CALL(*subscriber, onComplete_());
+
+  f->subscribe(subscriber);
+  EXPECT_EQ(v, std::vector<int64_t>({1, 2, 3, 4, 5}));
+
+  subscriber->subscription()->request(5);
+  EXPECT_EQ(v, std::vector<int64_t>({1, 2, 3, 4, 5, 6, 7, 8, 9}));
+}
+
+TEST(Observable, toFlowableLatestStrategy) {
+  auto a = Observables::range(1,10);
+  auto f = a->toFlowable(BackpressureStrategy::LATEST);
+
+  std::vector<int64_t> v;
+
+  auto subscriber = make_ref<testing::StrictMock<yarpl::flowable::MockSubscriber<int64_t>>>(5);
+
+  EXPECT_CALL(*subscriber, onSubscribe_(_));
+  EXPECT_CALL(*subscriber, onNext_(_))
+      .WillRepeatedly(Invoke([&](int64_t value) {
+          v.push_back(value);
+      }));
+  EXPECT_CALL(*subscriber, onComplete_());
+
+  f->subscribe(subscriber);
+  EXPECT_EQ(v, std::vector<int64_t>({1, 2, 3, 4, 5}));
+
+  subscriber->subscription()->request(5);
+  EXPECT_EQ(v, std::vector<int64_t>({1, 2, 3, 4, 5, 9}));
```

Comment:
So LATEST is like DROP but we buffer only the last element?

Reply:
that is correct: http://reactivex.io/RxJava/2.x/javadoc/io/reactivex/BackpressureStrategy.html